enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,195 @@
+  Exception granted to run that deprecated version of OpenStack.
+
+### Non-Goals
+
+This enhancement does not intent do block any OpenShift operations done
+outside of the installer if the OpenStack version is not supported.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made
+in the openshift/installer:
+
+- Update gophercloud vendoring to (at least) [b56bf1](https://github.com/gophercloud/gophercloud/pull/2037/commits/5a1daf082451587459058cc539c53f2ae1aaaef3)
+  so we can beneficiate from the apiversions package.
+- Create a constant that will list all the supported versions
+  of Compute API based on the upstream manual.
+- Add a new validation against the OpenStack cloud to make sure that
+  the Compute API version is supported.
+- Create a configuration flag `skipVersionCheck` that when set to
+   false will disable the OpenStack version check and proceed anyway.
+
+### Design Details
+
+The OpenStack Compute service (also known as Nova) is always deployed
+in OpenStack clouds. Also, it is one of the services that bumps the
+microversions so regularly that is it possible to figure out
+what version of OpenStack is running based on the maximum version of
+Nova API that is available for a cloud.
+
+Therefore, it is proposed that we use that information as a reliable
+way of checking if the version of OpenStack that is running is
+actually supported.
+
+The Nova API versions that are supported will be maintained in
+the installer code and will follow the recommendations from
+the [upstream Nova API manual](https://docs.openstack.org/nova/latest/reference/api-microversion-history.html).
+
+There are multiple reasons where an administrator would want to skip
+the OpenStack version check:
+- the administrators know what they are doing and understand that at this
+  point they aren't supported anymore.
+- the administrators have performed a major upgrade or Fast-Forward-Upgrade
+  of OpenStack and the operation is partially finished; however
+  they need to proceed to an OpenShift operation (e.g. scale up).
+  It it possible to run an OpenStack cloud with various versions of
+  APIs, as long as compatibility constraints are respected (out of
+  scope for this document).
+- the administrators have a Support Exception and an OpenStack upgrade
+  isn't possible at the moment of the operation.
+
+In that case, we would use a new configuration flag named `skipVersionCheck`.
+
+### User Stories
+
+#### Check the OpenStack version
+
+As an enterprise OpenShift cluster administrator, I want to make
+sure that the version of OpenStack that will run or is already
+running the cluster, is actually supported. If not, I would like
+to see the installer to fail early.
+
+#### Skip the version check
+
+As an enterprise OpenShift cluster administrator, I know what I'm doing
+and need to skip the version check in a case where I absolutely need
+to run an Openshift operation (e.g. scale).
```

Comment:
I'm not sure I'd add the version check to any operation which doesn't change the version of OCP. I think performing a scale operation on an unsupported openstack would require one of:

1. The underlying OpenStack has been downgraded
2. OCP has removed support for the underlying OpenStack
3. The underlying OpenStack has been upgraded to an unsupported version

I don't think we support (1). I don't think we would ever do (2). (3) would imply an OpenStack regression.

Everything else being correct this should just be a sanity check so it shouldn't hurt, but it does add a potential source of failure.

I would be inclined to add this check only to install and upgrade.

Reply:
thanks for this nice comment, I added it to the text, it helps a lot to understand the context.